bundles/azure/aso/psql/bundle/config/01-flexible-server.ytt.yml

```diff
@@ -0,0 +1,248 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:assert", "assert")
+#@ load("@ytt:regexp", "regexp")
+#@ load("@ytt:ip", "ip")
+
+
+#@ flexibleServerName    = data.values.name
+#@ dbSecretName          = data.values.name
+#@ resourceGroupK8SName  = data.values.name
+#@ serviceAccountName      = data.values.name + "-reader"
+#@ serviceAccountNamespace = data.values.namespace
+#@ roleName                = data.values.name + "-reading"
+
+#@ mandatory_fields = [ "name", "namespace", "location", "firewall_rules", "aso_controller_namespace" ]
+#@ missing_fields = []
+#@ for field in mandatory_fields:
+#@   if len(data.values[field]) == 0:
+#@     missing_fields.append(field)
+#@   end
+#@ end
+
+#@ if data.values.database.name == "":
+#@   missing_fields.append("database.name")
+#@ end
+#@ if data.values.resource_group.name == "":
+#@   missing_fields.append("resource_group.name")
+#@ end
+#@ if len(missing_fields) > 0:
+#@   assert.fail("Missing values for mandatory fields '{}'".format(missing_fields))
+#@ end
+
+#@ if not regexp.match("^Standard_", data.values.server.instance_type):
+#@   assert.fail("'instance_type' must follow the convention Standard_(VMname) (was: '{}')".format(data.values.server.instance_type))
+#@ end
+
+#@ if data.values.server.instance_tier not in ["Burstable", "GeneralPurpose", "Memory Optimized"]:
+#@   assert.fail("'instance_tier' must be in [\"Burstable\", \"GeneralPurpose\", \"Memory Optimized\"] (was: '{}')".format(data.values.server.instance_tier))
+#@ end
+
+#! @ if data.values.server.instance_storage_size_gb < 32 or data.values.server.instance_storage_size_gb > 16384:
+#! @   assert.fail("'instance_storage_size_gb' must be within 32 and 16384 (was: {})".format(data.values.server.instance_storage_size_gb))
+#! @ end
+
+#@ if data.values.create_namespace:
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: #@ data.values.namespace
+#@ end
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: Password
+metadata:
+  name: #@ dbSecretName
+  namespace: #@ data.values.namespace
+spec:
+  length: 64
+  secretTemplate:
+    type: Opaque
+    stringData:
+      password: $(value)
+---
+apiVersion: resources.azure.com/v1beta20200601
+kind: ResourceGroup
+metadata:
+  name: #@ resourceGroupK8SName
+  namespace: #@ data.values.namespace
+  annotations:
+    serviceoperator.azure.com/operator-namespace: #@ data.values.aso_controller_namespace
+#@ if/end data.values.resource_group.use_existing:
+    serviceoperator.azure.com/reconcile-policy: skip
+  labels:
+    kapp.k14s.io/noop: ""
+spec:
+  azureName: #@ data.values.resource_group.name
+  location: #@ data.values.location
+#@ tags_list = []
+#@ tags_list.extend(data.values.global_tags)
+#@ tags_list.extend(data.values.resource_group.tags)
+#@ tags = {x["key"]: x["value"] for x in tags_list}
+#@ if/end len(tags) > 0:
+  tags: #@ tags
```

Comment:
_nit_: shouldn't that be a function or something? DRY.

Reply:
Good point, to be continued